File: frontend/src/lib/api/meeting-documents.ts
/**
 * Meeting Agenda & Minutes API
 */

import { get, post, put, del } from "./helpers";

export interface AgendaItem {
  id: string;
  meeting_id: string;
  title: string;
  description: string | null;
  duration_minutes: number | null;
  presenter: { id: string; name: string; avatar_url: string | null } | null;
  sort_order: number;
  is_completed: boolean;
  created_by: { id: string; name: string };
  created_at: string;
  updated_at: string;
}

export interface MeetingMinutesData {
  id: string;
  meeting_id: string;
  content: string;
  summary: string | null;
  decisions: string | null;
  action_items: string | null;
  is_published: boolean;
  recorded_by: { id: string; name: string };
  created_at: string;
  updated_at: string;
}

const base = (eventId: string, meetingId: string) =>
  `/events/${eventId}/meetings/${meetingId}`;

export const meetingDocsApi = {
  // Agenda
  listAgenda: (eventId: string, meetingId: string) =>
    get(`${base(eventId, meetingId)}/agenda`),

  createAgendaItem: (eventId: string, meetingId: string, data: {
    title: string;
    description?: string;
    duration_minutes?: number;
    presenter_user_id?: string;
    sort_order?: number;
  }) => post(`${base(eventId, meetingId)}/agenda`, data),

  updateAgendaItem: (eventId: string, meetingId: string, itemId: string, data: {
    title?: string;
    description?: string;
    duration_minutes?: number;
    presenter_user_id?: string;
    sort_order?: number;
    is_completed?: boolean;
  }) => put(`${base(eventId, meetingId)}/agenda/${itemId}`, data),

  deleteAgendaItem: (eventId: string, meetingId: string, itemId: string) =>
    del(`${base(eventId, meetingId)}/agenda/${itemId}`),

  reorderAgenda: (eventId: string, meetingId: string, itemIds: string[]) =>
    post(`${base(eventId, meetingId)}/agenda/reorder`, { item_ids: itemIds }),

  // Minutes
  getMinutes: (eventId: string, meetingId: string) =>
    get(`${base(eventId, meetingId)}/minutes`),

  createMinutes: (eventId: string, meetingId: string, data: {
    content: string;
    summary?: string;
    decisions?: string;
    action_items?: string;
  }) => post(`${base(eventId, meetingId)}/minutes`, data),

  updateMinutes: (eventId: string, meetingId: string, data: {
    content?: string;
    summary?: string;
    decisions?: string;
    action_items?: string;
    is_published?: boolean;
  }) => put(`${base(eventId, meetingId)}/minutes`, data),

  deleteMinutes: (eventId: string, meetingId: string) =>
    del(`${base(eventId, meetingId)}/minutes`),
};

File: backend/app/models/meeting_documents.py
"""
Meeting Agenda & Minutes Models
Supports agenda items and meeting minutes/notes for event meetings.
"""

from sqlalchemy import Column, Boolean, ForeignKey, DateTime, Text, Integer, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from core.base import Base


class MeetingAgendaItem(Base):
    __tablename__ = 'meeting_agenda_items'

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=func.gen_random_uuid())
    meeting_id = Column(UUID(as_uuid=True), ForeignKey('event_meetings.id', ondelete='CASCADE'), nullable=False)
    title = Column(Text, nullable=False)
    description = Column(Text)
    duration_minutes = Column(Integer)
    presenter_user_id = Column(UUID(as_uuid=True), ForeignKey('users.id', ondelete='SET NULL'))
    sort_order = Column(Integer, default=0)
    is_completed = Column(Boolean, default=False)
    created_by = Column(UUID(as_uuid=True), ForeignKey('users.id', ondelete='CASCADE'), nullable=False)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    # Relationships
    meeting = relationship("EventMeeting", backref="agenda_items")
    presenter = relationship("User", foreign_keys=[presenter_user_id])
    creator = relationship("User", foreign_keys=[created_by])


class MeetingMinutes(Base):
    __tablename__ = 'meeting_minutes'

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=func.gen_random_uuid())
    meeting_id = Column(UUID(as_uuid=True), ForeignKey('event_meetings.id', ondelete='CASCADE'), nullable=False, unique=True)
    content = Column(Text, nullable=False)  # Rich text / markdown
    summary = Column(Text)
    decisions = Column(Text)  # Key decisions made
    action_items = Column(Text)  # Action items from the meeting
    recorded_by = Column(UUID(as_uuid=True), ForeignKey('users.id', ondelete='CASCADE'), nullable=False)
    is_published = Column(Boolean, default=False)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    # Relationships
    meeting = relationship("EventMeeting", backref="minutes")
    recorder = relationship("User", foreign_keys=[recorded_by])
